scenarios/command/c/command_server/main.c

```diff
@@ -4,36 +4,61 @@
-  do                                                                  \
-  {                                                                   \
-    if (rc != MOSQ_ERR_SUCCESS)                                       \
-    {                                                                 \
-      printf("Error sending response: %s\n", mosquitto_strerror(rc)); \
-      free(response_topic);                                           \
-      response_topic = NULL;                                          \
-      free(correlation_data);                                         \
-      correlation_data = NULL;                                        \
-      mosquitto_property_free_all(&response_props);                   \
-      response_props = NULL;                                          \
-      return;                                                         \
-    }                                                                 \
+#define RETURN_IF_ERROR(rc)                                                           \
+  do                                                                                  \
+  {                                                                                   \
+    if (rc != MOSQ_ERR_SUCCESS)                                                       \
+    {                                                                                 \
+      printf("[ERROR] Failure while sending response: %s\n", mosquitto_strerror(rc)); \
+      free(response_topic);                                                           \
+      response_topic = NULL;                                                          \
+      free(correlation_data);                                                         \
+      correlation_data = NULL;                                                        \
+      mosquitto_property_free_all(&response_props);                                   \
+      response_props = NULL;                                                          \
+      free(payload_buf);                                                              \
+      payload_buf = NULL;                                                             \
+      return;                                                                         \
+    }                                                                                 \
   } while (0)
 
+// Function to execute unlock request. For this sample, it just prints the request information.
+bool handle_unlock(char* payload, int payload_length)
+{
+  UnlockRequest* unlock_request = unlock_request__unpack(NULL, payload_length, payload);
+  if (unlock_request == NULL)
+  {
+    printf("\t[ERROR] Failure deserializing protobuf payload\n");
+    return false;
+  }
+  else
+  {
+    printf(
+        "\tUnlock request sent from %s at %s",
+        unlock_request->requestedfrom,
+        asctime(localtime(&unlock_request->when->seconds)));
```

Comment:
The vehicle should verify the request time:
1. the request clock was authentic as well as in-sync with GPS/Cell
2. the request isn't expired - door should not unlock if the original time was more than 5 seconds ago (requestor app should clearly wait for 5 seconds to indicate that there is a possibility that the door can open).
 
Also, if other requests arrive in a short burst, the code should reconcile the requests (if any is "lock", the door should fail-safe to locked).

Reply:
Resolving with work item opened from my comment below